client/client.go

```diff
@@ -349,11 +349,12 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
+		resp.Body.Close()
 		<-done
-		if err == nil {
-			err = ctx.Err()
-		}
+		// Because the body is closed, the err var from ReadAll goroutine is
```

Comment:
Yes, the comment is not correct, and it could return nil error.

I think the behavior is acceptable. If user cancels, he has the expectation that he will receive cancel error. Or if user cancels a finished request, it should have gone through `case <-done` yet because it is finished. The probability that two select cases are ready at the same time is low, so it affects little.

It is better to check whether the error matches its expectation, and replaces it only when it matches. But the error check does not have a standard way because it relies on the error string, which has no guarantee to keep the same.